File: YangDoPlatform/app/src/main/java/se/jbnu/yangdoplatform/Fragment/Fragment_Setting.java
package se.jbnu.yangdoplatform.Fragment;

import android.content.DialogInterface;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.preference.PreferenceManager;

import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.Fragment;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;

import se.jbnu.yangdoplatform.HomeActivity;
import se.jbnu.yangdoplatform.R;
import se.jbnu.yangdoplatform.SigninActivity;
import se.jbnu.yangdoplatform.SplashActivity;


/**
 * A simple {@link Fragment} subclass.
 * Use the {@link Fragment_Setting#newInstance} factory method to
 * create an instance of this fragment.
 */
public class Fragment_Setting extends PreferenceFragmentCompat {
    SharedPreferences prefs;

    public static Fragment_Setting newInstance() {
        return new Fragment_Setting();
    }

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        addPreferencesFromResource(R.xml.settings_preference);
        prefs = PreferenceManager.getDefaultSharedPreferences(getActivity());

        androidx.preference.Preference buttonLogout = (androidx.preference.Preference) findPreference("Logout");
        androidx.preference.Preference buttonAccountDelete = (androidx.preference.Preference) findPreference("AccountDelete");


        buttonLogout.setOnPreferenceClickListener(new androidx.preference.Preference.OnPreferenceClickListener() {
            public boolean onPreferenceClick(androidx.preference.Preference preference) {
                //open browser or intent here
                SigninActivity signinActivity = new SigninActivity();
                signinActivity.signOut();
                Intent intent = new Intent(getActivity().getApplicationContext(), SplashActivity.class);
                startActivity(intent);
                return true;
            }
        });

        buttonAccountDelete.setOnPreferenceClickListener(new androidx.preference.Preference.OnPreferenceClickListener() {
            public boolean onPreferenceClick(Preference preference) {
                //open browser or intent here
                AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
                builder.setTitle("회원탈퇴");
                builder.setMessage("정말 회원탈퇴를 진행하시겠습니까?");
                builder.setPositiveButton("OK", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                            SigninActivity signinActivity = new SigninActivity();
                            signinActivity.AccountDelete();
                            Intent intent = new Intent(getActivity().getApplicationContext(), SplashActivity.class);
                            startActivity(intent);
                    }
                });
                builder.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialog, int which) {
                        ((HomeActivity)getActivity()).replaceFragment(Fragment_Setting.newInstance());
                    }
                });
                builder.create().show();
                return true;
            }
        });
    }


}

File: YangDo_Platform/app/src/main/java/se/jbnu/yangdoplatform/Fragment/Fragment_Setting.kt
package se.jbnu.yangdoplatform.Fragment

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.preference.PreferenceManager
import androidx.appcompat.app.AlertDialog
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import se.jbnu.yangdoplatform.HomeActivity
import se.jbnu.yangdoplatform.R
import se.jbnu.yangdoplatform.SigninActivity
import se.jbnu.yangdoplatform.SigninActivity.Companion.AccountDelete
import se.jbnu.yangdoplatform.SigninActivity.Companion.signOut
import se.jbnu.yangdoplatform.SplashActivity

/**
 * A simple [Fragment] subclass.
 * Use the [Fragment_Setting.newInstance] factory method to
 * create an instance of this fragment.
 */
class Fragment_Setting : PreferenceFragmentCompat() {
    var prefs: SharedPreferences? = null
    override fun onCreatePreferences(savedInstanceState: Bundle, rootKey: String) {
        addPreferencesFromResource(R.xml.settings_preference)
        prefs = PreferenceManager.getDefaultSharedPreferences(activity)
        val buttonLogout = findPreference("Logout") as Preference?
        val buttonAccountDelete = findPreference("AccountDelete") as Preference?
        buttonLogout!!.onPreferenceClickListener = Preference.OnPreferenceClickListener { //open browser or intent here
            val signinActivity = SigninActivity()
            signOut()
            val intent = Intent(requireActivity().applicationContext, SplashActivity::class.java)
            startActivity(intent)
            true
        }
        buttonAccountDelete!!.onPreferenceClickListener = Preference.OnPreferenceClickListener { //open browser or intent here
            val builder = AlertDialog.Builder(requireActivity())
            builder.setTitle("회원탈퇴")
            builder.setMessage("정말 회원탈퇴를 진행하시겠습니까?")
            builder.setPositiveButton("OK") { dialog, which ->
                val signinActivity = SigninActivity()
                AccountDelete()
                val intent = Intent(requireActivity().applicationContext, SplashActivity::class.java)
                startActivity(intent)
            }
            builder.setNegativeButton("Cancel") { dialog, which -> (activity as HomeActivity?)!!.replaceFragment(newInstance()) }
            builder.create().show()
            true
        }
    }

    companion object {
        @JvmStatic
        fun newInstance(): Fragment_Setting {
            return Fragment_Setting()
        }
    }
}